### PR TITLE
Fix/sheet run animation after initial position is set

### LINKED ISSUE #2506: Sheet bounces on mount when default open
### Current Behavior

When having a sheet be default open in a component, it bounces around on mount. 

https://github.com/tamagui/tamagui/assets/1610241/d168c2e6-0516-40c8-a04e-98f658748293

### Expected Behavior

The sheet should open to the specified position instead of yoyo-ing up and down into position.

### Tamagui Version

```markdown
^1.94.1
```

### Platform (Web, iOS, Android)

```markdown
iOS (simulator)
```

### Reproduction

```markdown
Repo https://github.com/wollerman/tamagui-sheet-bug-demo/commits/main/

specific commmit adding the sheet: https://github.com/wollerman/tamagui-sheet-bug-demo/commit/2dfbf44f5e36560dc69524bd385f01cb4fa966c0
```

### System Info

```markdown
❯ npx envinfo --system --npmPackages --binaries --browsers

  System:
    OS: macOS 14.4
    CPU: (8) arm64 Apple M1 Pro
    Memory: 88.78 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    Yarn: 4.0.2 - ~/.nvm/versions/node/v20.11.0/bin/yarn
    npm: 10.4.0 - ~/.nvm/versions/node/v20.11.0/bin/npm
    Watchman: 2024.04.01.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 123.0.6312.106
    Safari: 17.4
  npmPackages:
    @babel/core: ^7.23.3 => 7.23.7
    @expo/metro-runtime: ~3.1.3 => 3.1.3
    @react-navigation/native: ^6.0.2 => 6.1.9
    @tamagui/config: ^1.94.1 => 1.94.1
    @tamagui/metro-plugin: ^1.94.1 => 1.94.1
    @types/react: ~18.2.14 => 18.2.47
    babel-preset-expo: ^10.0.1 => 10.0.1
    expo: ~50.0.6 => 50.0.6
    expo-font: ~11.10.2 => 11.10.2
    expo-linking: ~6.2.1 => 6.2.2
    expo-router: ~3.4.7 => 3.4.7
    expo-splash-screen: ~0.26.4 => 0.26.4
    expo-status-bar: ~1.11.1 => 1.11.1
    expo-system-ui: ~2.9.3 => 2.9.3
    expo-web-browser: ~12.8.2 => 12.8.2
    react: ^18.2.0 => 18.2.0
    react-dom: ^18.2.0 => 18.2.0
    react-helmet-async: ~2.0.4 => 2.0.4
    react-native: 0.73.4 => 0.73.4
    react-native-reanimated: ~3.6.2 => 3.6.2
    react-native-safe-area-context: 4.8.2 => 4.8.2
    react-native-screens: ~3.29.0 => 3.29.0
    react-native-web: ^0.19.10 => 0.19.10
    tamagui: ^1.94.1 => 1.94.1
    typescript: ^5.3.3 => 5.3.3
```
```

### PR DESCRIPTION
**Pull request by TheEhsanSarshar**
fix #2506

### CODE REVIEW

File: packages/sheet/src/SheetImplementationCustom.tsx [hunk 1]
```diff
@@ -175,17 +175,27 @@ export const SheetImplementationCustom = themeable(
...
-          duration: 0,
-        })
+        animatedNumber.setValue(
+          screenSize,
+          {
+            type: 'timing',
+            duration: 0,
+          },
+          () => {
+            if (isHidden) {
```
Comment (natew) [inline packages/sheet/src/SheetImplementationCustom.tsx:185]: do we need to support this callback functionality in css + native animations too?

File: packages/animations-css/src/createAnimations.tsx [hunk 2]
```diff
@@ -24,12 +24,13 @@ export function createAnimations<A extends Object>(animations: A): AnimationDriv
...
           return val
         },
-        setValue(next) {
+        setValue(next, config, onFinish) {
           setVal(next)
           const listeners = reactionListeners.get(setVal)
           if (listeners) {
             listeners.forEach((cb) => cb(next))
           }
+          onFinish?.()
```
Comment (natew) [inline packages/animations-css/src/createAnimations.tsx:33]: don't we need to listen for `transitionend`?

File: packages/sheet/src/SheetImplementationCustom.tsx (hunk 1 above)
Comment (TheEhsanSarshar) [reply packages/sheet/src/SheetImplementationCustom.tsx:185]: we don't need it for CSS, but just to have the same API, having onFinish for CSS is fine I guess?

**Comment by cavishek39** [comment]
What is soultion of this issue? Since, I'm still facing this issue

**Pull request merged**